### ISSUE TITLE
Action ALWAYS reports "The following file(s) are tracked in LFS and were likely accidentally checked in:" for PDF files even if properly stored in LFS

### ISSUE THREAD

**Issue by tsimbalar** [ActionsDesk/lfs-warning#135]
**Describe the bug**
For Pull Requests that contain pdf files, even when the pdf is properly added through LFS, the action reports : 

```
## :warning: Possible file(s) that should be tracked in LFS detected :warning:
        
        The following file(s) are tracked in LFS and were likely accidentally checked in:

        filename.pdf
      
```

### Initial state
1. In a git repo with LFS properly configured, and `.gitattributes` such as 
 ```
*.png filter=lfs diff=lfs merge=lfs -text
*.pdf filter=lfs diff=lfs merge=lfs -text
*.woff filter=lfs diff=lfs merge=lfs -text
*.woff2 filter=lfs diff=lfs merge=lfs -text
*.ico filter=lfs diff=lfs merge=lfs -text

 ```
2. with a GitHub Actions workflow like : 
```yaml
name: LFS-Checker

on:
  pull_request:
  workflow_dispatch: # to run it manually

jobs:
  lfs-checking:
    runs-on: ubuntu-18.04
    steps:
      - name: Checkout repo
        uses: actions/checkout@v2.3.5
        with:
          lfs: false

      - name: LFS Warning
        uses: actionsdesk/lfs-warning@v3.1
```

### Repro steps:
1. Create a branch, and add a pdf file in it, properly adding it to LFS
2. Push the branch and create a PR
3. Wait for the action workflow to be executed

### Observed behavior:

A comment is added to the PR, and the action is marked as failed : 

![image](https://user-images.githubusercontent.com/160544/161950361-d769b7e2-86d1-4404-b271-6be526b8df2f.png)

## Expected behavior
There should not be any warning, and the "PR-check" should be green, given that the pdf file is properly added in LFS.

## Screenshots

GitHub properly reports that the file is stored in LFS when navigating to the file: 
![image](https://user-images.githubusercontent.com/160544/161950860-952ff548-6833-4474-b94d-7f786312357f.png)

## Additional context
With my colleague @davidlj95 we tried to investigate a bit more and noticed a few things : 

- GitHub UI : **Different behavior for `.pdf` files**, compared to other binary files like `.mp4` or `.png`

In the GitHub UI, in a PR that contains `.mp4` or `.png` files added to LFS, the GitHub PR `file` view shows "Git LFS file not shown" : 

![image](https://user-images.githubusercontent.com/160544/161951523-494ee674-6a76-49bc-a538-8880b3514973.png)

![image](https://user-images.githubusercontent.com/160544/161951651-009eaef4-8067-4223-bfe3-259ada486b51.png)

For PDFs, instead it shows "Binary file not shown" : 

![image](https://user-images.githubusercontent.com/160544/161951675-ce17c380-a8e6-4cb5-bd9d-0cf53e8dedef.png)

- GitHub API : **`.pdf` files have no `patch` property** in the response to `GET https://api.github.com/repos/{org}/{repo}/pulls/{pr-id}}/files`

In a PR that adds a `png` and a `pdf` file (added in LFS), 

```curl
curl -H "Authorization: token MY_SECRET_TOKEN" -X GET https://api.github.com/repos/koa-health/mindset-website/pulls/39/files
```

returns

```json
[
  {
    "sha": "256a6025de6227c29ab634ff8f85affa413d72c2",
    "filename": "website/dummy.pdf",
    "status": "added",
    "additions": 0,
    "deletions": 0,
    "changes": 0,
    "blob_url": "https://github.com/koa-health/mindset-website/blob/e1ade945635fdad92b716e1f1aba9b7b8f4d1778/website%2Fdummy.pdf",
    "raw_url": "https://github.com/koa-health/mindset-website/raw/e1ade945635fdad92b716e1f1aba9b7b8f4d1778/website%2Fdummy.pdf",
    "contents_url": "https://api.github.com/repos/koa-health/mindset-website/contents/website%2Fdummy.pdf?ref=e1ade945635fdad92b716e1f1aba9b7b8f4d1778"
  },
  {
    "sha": "68b76bda66713a7c945ee4fdb4c470b639dbe3c9",
    "filename": "website/dummy.png",
    "status": "added",
    "additions": 3,
    "deletions": 0,
    "changes": 3,
    "blob_url": "https://github.com/koa-health/mindset-website/blob/e1ade945635fdad92b716e1f1aba9b7b8f4d1778/website%2Fdummy.png",
    "raw_url": "https://github.com/koa-health/mindset-website/raw/e1ade945635fdad92b716e1f1aba9b7b8f4d1778/website%2Fdummy.png",
    "contents_url": "https://api.github.com/repos/koa-health/mindset-website/contents/website%2Fdummy.png?ref=e1ade945635fdad92b716e1f1aba9b7b8f4d1778",
    "patch": "@@ -0,0 +1,3 @@\n+version https://git-lfs.github.com/spec/v1\n+oid sha256:fa3be2769d3074bc9c5a31a9039f55c9109988e1b3ea4e99731058b0497146e0\n+size 54967"
  }
]
```

note how the entry for the `png` file contains a `patch` property, with the expected "flag" for LFS ... but the `pdf` entry simply has no `patch` property. 

## Probable Root cause of the issue
Based on lack of `patch` property, this part of the code is misbehaving : 

https://github.com/ActionsDesk/lfs-warning/blob/6090ed546050b18e4341e8decc0805d33572fb65/src/index.ts#L52-L55

and `isStoredInLFS` ends up as `false` for PDFs, even if they are properly stored.

## Possible fix
For PDFs, the `patch` property is missing, so we can't rely on it to know whether the "lfs flag" is set or not ... but this information can be obtained through other means. 

Given the response to the "get files for pull request" API call (see above), we have the `sha` of the commit ... from there we can use the "git Blob" API to retrieve the corresponsing contents of the file. 

(it turns out in the current implementation of the action, we are already fetching this information : 

https://github.com/ActionsDesk/lfs-warning/blob/6090ed546050b18e4341e8decc0805d33572fb65/src/index.ts#L184

)

If we do that API call, we get something like : 

```
curl -H "Authorization: token MY_SECRET_TOKEN" -X GET https://api.github.com/repos/koa-health/mindset-website/git/blobs/256a6025de6227c29ab634ff8f85affa413d72c2
```

```json
{
  "sha": "256a6025de6227c29ab634ff8f85affa413d72c2",
  "node_id": "MDQ6QmxvYjQwMzU2MDc5MToyNTZhNjAyNWRlNjIyN2MyOWFiNjM0ZmY4Zjg1YWZmYTQxM2Q3MmMy",
  "size": 130,
  "url": "https://api.github.com/repos/koa-health/mindset-website/git/blobs/256a6025de6227c29ab634ff8f85affa413d72c2",
  "content": "dmVyc2lvbiBodHRwczovL2dpdC1sZnMuZ2l0aHViLmNvbS9zcGVjL3YxCm9p\nZCBzaGEyNTY6M2RmNzlkMzRhYmJjYTk5MzA4ZTc5Y2I5NDQ2MWMxODkzNTgy\nNjA0ZDY4MzI5YTQxZmQ0YmVjMTg4NWU2YWRiNApzaXplIDEzMjY0Cg==\n",
  "encoding": "base64"
}
```

And if we look at the content end decode it (base 64 decode)

```
> echo 'dmVyc2lvbiBodHRwczovL2dpdC1sZnMuZ2l0aHViLmNvbS9zcGVjL3YxCm9p\nZCBzaGEyNTY6M2RmNzlkMzRhYmJjYTk5MzA4ZTc5Y2I5NDQ2MWMxODkzNTgy\nNjA0ZDY4MzI5YTQxZmQ0YmVjMTg4NWU2YWRiNApzaXplIDEzMjY0Cg==\n' | base64 -d

version https://git-lfs.github.com/spec/v1
oid sha256:3df79d34abbca99308e79cb94461c1893582604d68329a41fd4bec1885e6adb4
size 13264
```

we can retrieve the necessary information to know that this file is indeed stored in LFS.

So it seems like a possible approach to address this issue is to either : 
- no longer look at the `patch` property, and instead ALWAYS look at the `content` of the blob
- or ... use the `patch` property when it's available, and if it's not, look at the content of the blob

I hope this helps :)

**Comment by ppremk** [comment]
👋🏽 @tsimbalar thank you for reporting this bug. I agree that in the initial version of this action, we didn't factor in working with LFS tracked files. This is an important catch! 🙇🏽 

In the spirit of building a community around this and facilitating for more contribution, would you be open to submit a pull request with the suggested fix? It will help expedite a fix and we can cut a new release ❤️ 🙂 

cc/ @naseemkullah

**Comment by davidlj95** [comment]
Hi @ppremk , thanks for your response ! 

Proposed PR for the fix in #138
